KIPs/kip-237.md

```diff
@@ -0,0 +1,75 @@
+---
+kip: 237
+title: Storage Compression
+author: Lake (@hyunsooda), Ollie (@blukat29)
+discussions-to: TBD
+status: Draft
+category: Storage
+created: 2024-12-26
+---
+
+## Abstract
+Compress the block header, body, and receipts data in its persistent form to reduce disk usage.
+With an efficient data structure, historic data remains accessible with minimal query overhead from the lossless-compressed database.
+
+## Motivation
+Kaia has been focusing on optimizing the state trie storage through [state migration and live pruning](https://docs.kaia.io/learn/storage/state-pruning/).
+While these optimizations dealt with the most significant part of the database, there is still potential to further reduce the other parts of the database.
+This KIP reduces the storage footprint of the following three data types:
+- Header: RLP-encoded block header. Most header fields are repeated across headers.
+- Body: RLP-encoded transaction bodies in a block. Transaction calldatas tend to contain many zero bytes and the contents are often similar between different transactions.
+- Receipts: RLP-encoded transaction receipts in a block. Receipt event logs tend to contain many zero bytes and the contents are often similar between different transactions.
+
+The three data types are generated every time a new block is inserted.
+They consume a substantial amount of storage. As of 2024-12-02, In the migrated mainnet chaindata, the header, body, and receipts account for 10.4%, 28.1%, and 40.9% of the total size, respectively, which together occupy 79.4% of the node's disk space.
+This illustrates that the state trie is no longer a dominant data in the full-node with pruned states.
+This KIP proposes to compress the three data types to reduce the total disk size, thereby making the node operation more efficient and affordable.
+
+### Compression
+A node periodically compresses storage in a background thread and removes the original data once it is confirmed safe to do so.
+Data is compressed in units called "chunks".
+An optimal chunk size should balance a high compression rate with minimal decompression overhead.
+Determining the ideal chunk capacity remains an open problem. However, our experiments suggest that a 1MB chunk size offers a favorable trade-off.
```

Comment:
Maybe we should state which compression algorithm is being used.

Reply:
https://github.com/kaiachain/kips/pull/37/commits/1e97d19cfaec0466b73f2366eb4313a0ddc8c927

Reply:
Where did you mention about the zstd algorithm? Stating the exact algorithm (with the exact strength setting) can help understanding the exact database schema.

Reply:
I'm sorry, missed this. Added.